lib/runner.js

```diff
@@ -53,14 +53,25 @@ Runner.prototype.addSerialTest = function (title, cb) {
 };
 
 Runner.prototype.addBeforeHook = function (title, cb) {
-	this.tests.before.push(new Test(title, cb));
+	var test = new Test(title, cb);
+	test.type = 'hook';
```

Comment:
Yeah, we can look into this later. It could be a subclass too, `new Hook()`, but I'm not a big fan of that either. Maybe using Symbols as constants. `ava.HOOK_TYPE = Symbol('hook')` or something. But meh.